scripts/zones/Bastok_Markets/npcs/Reinberta.lua

```diff
@@ -22,12 +22,14 @@ function onTrade(player, npc, trade)
             player:setSkillRank(tpz.skill.GOLDSMITHING, newRank)
             player:startEvent(301, 0, 0, 0, 0, newRank, 1)
             player:setCharVar("GoldsmithingExpertQuest",2)
+            player:tradeComplete()
```

Comment:
when using tradeComplete or confirming   traded, that should _always_ happen in the events finish code, not inside the trade code where the event started. Here you have the trade complete while the event is running and if the player is disconnected the event code never completed but the item is gone.

Reply:
Moved `tradeComplete()` from `onTrade` to `onEventComplete`